github/src/main/kotlin/gropius/sync/github/IMSConfigManager.kt

```diff
@@ -0,0 +1,156 @@
+
+    /**
+     * Template fields for IMSTemplate and IMSProjectTemplate
+     */
+    private val commonTemplateFields = mapOf("bot-user" to obj {
+        "\$schema" to schema
+        "type" to arr["null", "string"]
+    }.toString(), "last-notification" to obj {
+        "\$schema" to schema
+        "type" to arr["null", obj {
+            "type" to "object"
+            "properties" to obj {
+                "title" to obj {
+                    "type" to "string"
+                }
+                "content" to obj {
+                    "type" to "string"
+                }
+            }
+            "required" to arr["title", "content"]
+            "gropius-type" to "notification"
+        }]
+    }.toString())
+
+    /**
+     * Name of the requested IMSTemplate
+     */
+    private val imsTemplateName = "Github"
+
+    /**
+     * Fields of the requestes IMSTemplate
+     */
+    private val imsTemplateFields = mapOf("read-user" to obj {
+        "\$schema" to schema
+        "type" to arr["null", obj {
+            "type" to "string"
+            "gropius-node" to "GropiusUser"
+            "gropius-type" to "github-user"
+        }]
+    }.toString()) + commonTemplateFields
+
+    /**
+     * Name of requested IMSProjectTemplate
+     */
+    private val imsProjectTemplateName = "Github"
+
+    /**
+     * Fields of the requestes IMSProjectTemplate
+     */
+    private val imsProjectTemplateFields = mapOf("repo" to obj {
+        "\$schema" to schema
+        "type" to "object"
+        "properties" to obj {
+            "owner" to obj {
+                "type" to "string"
+            }
+            "repo" to obj {
+                "type" to "string"
+            }
+        }
+        "required" to arr["owner", "repo"]
+        "gropius-type" to "github-owner"
+    }.toString()) + commonTemplateFields
+
+    /**
+     * Name of the ensured IMSIssueTemplate
+     */
+    private val imsIssueTemplateName = "Github Issue"
+
+    /**
+     * Fields of the required IMSIssueTemplate
+     */
+    private val imsIssueTemplateFields = mapOf<String, String>("last-notification" to obj {
+        "\$schema" to schema
+        "type" to arr["null", obj {
+            "type" to "object"
+            "properties" to obj {
+                "title" to obj {
+                    "type" to "string"
+                }
+                "content" to obj {
+                    "type" to "string"
+                }
+            }
+            "required" to arr["title", "content"]
+            "gropius-type" to "notification"
+        }]
+    }.toString())
+
+    /**
+     * Check if a given Template can be used as if it was the reference template.
+     * @param possibleTemplate a random template
+     * @param referenceName The title to check
+     * @param referenceType Check if all values specified by possibleTemplate are valid using the referenceType schemas
+     * @return True if valid
```

Comment:
maybe write `true` in lowercase, as you do not seem to capitalize everything

Reply:
97e7d2bad00bc5df055901a6122ebc81ba380105